tests/test_offline_preprocess.py

```diff
@@ -0,0 +1,149 @@
+#!/usr/bin/env python3
```

Comment:
Let’s discuss a better name? Maybe background? Offline implies we do it ahead of time instead of on the fly

Reply:
discussed offline, `background` is the way to go